src/main/java/com/fasterxml/jackson/core/filter/FilteringParserDelegate.java

```diff
@@ -418,7 +418,7 @@ public JsonToken nextToken() throws IOException
                 }
                 _itemFilter = f;
                 if (f == TokenFilter.INCLUDE_ALL) {
-                    if (_includePath) {
+                    if (_verifyAllowedMatches() && _includePath) {
```

Comment:
You're right, it would be faster, but it will not count properly anymore (please have a look at my previous comment)